nifi-weaver-processors/src/main/java/com/weaverplatform/nifi/individual/CreateIndividualProperty.java

```diff
@@ -0,0 +1,197 @@
+    .build();
+
+  public static final PropertyDescriptor SUBJECT_STATIC = new PropertyDescriptor
+    .Builder().name("Subject static")
+    .description("If there is no FlowFile attribute, use static value.")
+    .required(false)
+    .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+    .build();
+
+  public static final PropertyDescriptor PREDICATE_ATTRIBUTE = new PropertyDescriptor
+    .Builder().name("Predicate attribute")
+    .description("Look for the FlowFile attribute.")
+    .required(false)
+    .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+    .build();
+
+  public static final PropertyDescriptor PREDICATE_STATIC = new PropertyDescriptor
+    .Builder().name("Predicate static")
+    .description("If there is no FlowFile attribute, use static value.")
+    .required(false)
+    .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+    .build();
+
+  public static final PropertyDescriptor OBJECT_ATTRIBUTE = new PropertyDescriptor
+    .Builder().name("Object attribute")
+    .description("Look for the FlowFile attribute.")
+    .required(false)
+    .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+    .build();
+
+  public static final PropertyDescriptor OBJECT_STATIC = new PropertyDescriptor
+    .Builder().name("Object static")
+    .description("If there is no FlowFile attribute, use static value.")
+    .required(false)
+    .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+    .build();
+
+  public static final Relationship ORIGINAL = new Relationship.Builder()
+    .name("Original relationship")
+    .description("Original relationship to transfer content to.")
+    .build();
+
+  @Override
+  protected void init(final ProcessorInitializationContext context) {
+
+    super.init(context);
+
+    descriptors.add(SUBJECT_ATTRIBUTE);
+    descriptors.add(SUBJECT_STATIC);
+    descriptors.add(PREDICATE_ATTRIBUTE);
+    descriptors.add(PREDICATE_STATIC);
+    descriptors.add(OBJECT_ATTRIBUTE);
+    descriptors.add(OBJECT_STATIC);
+    this.properties = Collections.unmodifiableList(descriptors);
+
+    relationshipSet.add(ORIGINAL);
+    this.relationships = new AtomicReference<>(relationshipSet);
+  }
+
+  @Override
+  public void onTrigger(final ProcessContext context, final ProcessSession session) throws ProcessException {
+
+    super.onTrigger(context, session);
+
+    FlowFile flowFile = session.get();
+    if (flowFile == null) {
+      return;
+    }
+
+    String subject = null;
+    String subjectAttributeValue = context.getProperty(SUBJECT_ATTRIBUTE).getValue();
+    String subjectStaticValue    = context.getProperty(SUBJECT_STATIC).getValue();
+    if(subjectAttributeValue != null) {
+      subject = flowFile.getAttribute(subjectAttributeValue);
+    } else if(subjectStaticValue != null) {
+      subject = subjectStaticValue;
+    }
+    if(subject == null) {
+      throw new ProcessException("No subject found for this Individual Property.");
+    }
```

Comment:
The previous block of code looks similar to the next two blocks, and are straight up copy/paste from elsewhere. Extract similar code to a shared superclass or other utility methods for ease of maintenance.